api/pkg/apis/v1alpha1/vendors/catalogs-vendor.go

```diff
@@ -66,7 +66,7 @@ func (e *CatalogsVendor) Init(config vendors.VendorConfig, factories []managers.
 				if catalog.ParentName != "" {
 					catalog.ParentName = fmt.Sprintf("%s-%s", catalog.SiteId, catalog.ParentName)
 				}
-				err := e.CatalogsManager.UpsertSpec(context.TODO(), name, catalog)
+				err := e.CatalogsManager.UpsertSpec(context.TODO(), name, catalog, "")
```

Comment:
You need to set namespace for catalog somewhere like CatalogSpec.Metadata (or add Scope directly under CatalogSpec) at publisher side so you can extract the namespace here

Reply:
Do we want to save all synced catalog into default?